PalasoUIWindowsForms/Reporting/MemoryManagement.cs

```diff
@@ -52,30 +54,49 @@ public static bool CheckMemory(bool minor, string eventDescription, bool okToBot
 				var meminfo = File.ReadAllText("/proc/meminfo");
 				var match = new Regex(@"MemTotal:\s+(\d+) kB").Match(meminfo);
 				if (match.Success)
-				{
 					totalPhysicalMemory = ulong.Parse(match.Groups[1].Value) * 1024;
+				ulong totalSwapMemory = 0;
+				var match2 = new Regex(@"SwapTotal:\s+(\d+) kB").Match(meminfo);
+				if (match2.Success)
+					totalSwapMemory = ulong.Parse(match2.Groups [1].Value) * 1024;
+				var availableMemory = totalPhysicalMemory + totalSwapMemory;
+				if (is64BitProcess)
+				{
+					totalVirtualMemory = (availableMemory / 1024).ToString ("N0");
+				}
+				else
+				{
+					ulong twoGB = 2147483648L;
+					totalVirtualMemory = ((availableMemory > twoGB ? twoGB : availableMemory) / 1024).ToString("N0");
```

Comment:
Do you know for sure that this is the most VM a 32-bit .NET process can address on Linux? Anything up to 4G is conceivable.